### PR TITLE
Added Isrlu Function after cleaning up branch issues

### PR DESCRIPTION
**Pull request by PranavReddyP16**

### CODE REVIEW

**Comment by kartikdutt18** [comment]
Hi @PranavReddyP16, Could you add a description for this PR like why this function is useful or it's documentation.

**Comment by kartikdutt18** [comment]
Hi @PranavReddyP16, This still has mix up in branches.

**Comment by PranavReddyP16** [comment]
> Hi @PranavReddyP16, This still has mix up in branches.

Hey @kartikdutt18 I have completely deleted the old branch and created a new one with the same name after doing what you told me to erase all commits up till now. I did this so that there wouldn,t be any room for mistakes. Is there something else I should do?

**Comment by kartikdutt18** [comment]
Hi @PranavReddyP16, you ran the command from master, if I am not wrong. The issue is that you already have made changes in that branch. Repeat the commands again in this branch only then do `git push --force`

**Comment by PranavReddyP16** [comment]
@kartikdutt18 I'm facing a lot of issues. Can I do this:

completely delete both remote and local branches, create the branches again and then commit and push?

**Comment by kartikdutt18** [comment]
Please don't do that
In this branch run ` git reset --hard upstream/master`

**Comment by PranavReddyP16** [comment]
Ok I have done that

**Comment by kartikdutt18** [comment]
run `git push --force`

**Comment by PranavReddyP16** [comment]
When I do that I am getting this error:
fatal: The current branch isrlu has no upstream branch

And when I run the command:
git push --set-upstream origin isrlu

I get this error:
To https://github.com/PranavReddyP16/mlpack.git
 ! [rejected]            isrlu -> isrlu (non-fast-forward)
error: failed to push some refs to 'https://github.com/PranavReddyP16/mlpack.git'

followed by this:
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details

**Comment by kartikdutt18** [comment]
Use --force with push.

**Comment by PranavReddyP16** [comment]
I used git push --force

**Comment by kartikdutt18** [comment]
Did you try it after setting upstream?

**Comment by PranavReddyP16** [comment]
I followed these commands in order:
git remote add upstream https://github.com/mlpack/mlpack.git
git fetch upstream
git reset --hard upstream/master

**Comment by PranavReddyP16** [comment]
Wait when I run git remote add upstream https://github.com/mlpack/mlpack.git it says:
fatal: remote upstream already exists.

But when I run git remote -v it shows this:
origin  https://github.com/PranavReddyP16/mlpack.git (fetch)
origin  https://github.com/PranavReddyP16/mlpack.git (push)
upstream        https://github.com/mlpack/mlpack.git (fetch)
upstream        https://github.com/mlpack/mlpack.git (push)

So it's ok right?

**Comment by kartikdutt18** [comment]
Yes that's correct. Using git push --force should work (also try git add .
git commit -m "Forced Reset" and then git push)

**Comment by PranavReddyP16** [comment]
git push --force gives this output:
fatal: The current branch isrlu has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin isrlu

**Comment by kartikdutt18** [comment]
Run` git push --set-upstream origin isrlu --force`

**Comment by PranavReddyP16** [comment]
Ok I have run that command

**Comment by kartikdutt18** [comment]
It's even with master now(check online). Add the files and re-open this PR. (No need to create another branch you will face same issue).

**Comment by kartikdutt18** [comment]
Generally it's better to use master for rebasing.

**Comment by PranavReddyP16** [comment]
> Generally it's better to use master for rebasing.

Yes from now I will do that

File: src/mlpack/methods/ann/activation_functions/isrlu_function.hpp [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+   * activation y 
+   * @param alpha parameter, default value = 0.1
+   */
+  template<typename InputVecType, typename OutputVecType>
+  static void Deriv(const InputVecType& y,
+                    OutputVecType& x,
+                    const double alpha = 0.1)
+  {
+    if (y<0)
+      x = arma::pow(y / x, 3);
```
Comment (kartikdutt18) [inline src/mlpack/methods/ann/activation_functions/isrlu_function.hpp]: Use opening brackets here.

**Comment by PranavReddyP16** [comment]
> I think I will review it tomorrow, I hope that's okay.

No problem. Whenever you are free

**Pull request closed without merging**